### PR TITLE
initialize fevalues with NaN's to catch missing reinit!

### PR DESCRIPTION
**Pull request by fredrikekre**

### CODE REVIEW

File: src/FEValues/cell_values.jl [hunk 1]
```diff
@@ -65,8 +65,8 @@ function CellScalarValues(::Type{T}, quad_rule::QuadratureRule{dim, shape}, func
     # Function interpolation
     n_func_basefuncs = getnbasefunctions(func_interpol)
     N = zeros(T, n_func_basefuncs, n_qpoints)
-    dNdx = zeros(Vec{dim, T}, n_func_basefuncs, n_qpoints)
-    dNdξ = zeros(Vec{dim, T}, n_func_basefuncs, n_qpoints)
+    dNdx = [T(NaN)*zero(Vec{dim, T}) for i in 1:n_func_basefuncs, j in 1:n_qpoints]
+    dNdξ = [zero(Vec{dim, T}) for i in 1:n_func_basefuncs, j in 1:n_qpoints]
```
Comment (KristofferC) [inline src/FEValues/cell_values.jl]: use `fill`?

File: src/FEValues/cell_values.jl [hunk 2]
```diff
@@ -82,7 +82,7 @@ function CellScalarValues(::Type{T}, quad_rule::QuadratureRule{dim, shape}, func
         end
     end
 
-    detJdV = zeros(T, n_qpoints)
+    detJdV = scale!(Vector{T}(n_qpoints), T(NaN))
```
Comment (KristofferC) [inline src/FEValues/cell_values.jl]: use `fill`?

Comment (KristofferC) [reply src/FEValues/cell_values.jl]: etc for other places

File: src/FEValues/cell_values.jl (hunk 1 above)
Comment (fredrikekre) [reply src/FEValues/cell_values.jl]: `fill!(Matrix{Vec{dim,T}}(n_func_basefuncs, n_qpoints), NaN*zero(Vec{dim,T}))` ?

Comment (KristofferC) [reply src/FEValues/cell_values.jl]: `fill(NaN * zero(Vec{dim, T}), n_func_basefuncs, n_qpoints)`?

Comment (fredrikekre) [reply src/FEValues/cell_values.jl]: so smart

**Pull request merged**